resolver/serviceResolver.go

```diff
@@ -271,6 +271,24 @@ func (resolver *serviceResolver) SignMessage(userIp string, request requests.Sig
 
 }
 
+// SetSecurityModeNoExpire gets the user's guardian, verifies the codes and then sets the SecurityMode
+func (resolver *serviceResolver) SetSecurityModeNoExpire(userIp string, request requests.SetSecurityModeNoExpireMessage) (*requests.OTPCodeVerifyData, error) {
+	verifyCodeData, err := resolver.checkGuardianAndVerifyCode(userIp, request.UserAddr, request.Code, request.SecondCode)
+	if err != nil {
+		return verifyCodeData, err
+	}
+	return nil, resolver.secureOtpHandler.SetSecurityModeNoExpire(request.UserAddr)
```

Comment:
```suggestion
	return verifyCodeData, resolver.secureOtpHandler.SetSecurityModeNoExpire(request.UserAddr)
```
i would return verify code data here also, it's up to the caller to determine if it'll be used for the happy path or to ignore it

Reply:
updated